traffic_ops/testing/api/v4/parameters_test.go

```
@@ -231,3 +349,213 @@ func DeleteTestParameter(t *testing.T, pl tc.Parameter) {
+		}
+		if len(resp) <= 0 {
+			t.Errorf("No data available for Get Parameters by Config file")
+		}
+	}
+}
+
+func GetTestParametersByName(t *testing.T) {
+	for _, parameters := range testData.Parameters {
+		resp, reqInf, err := TOSession.GetParametersByName(parameters.Name, nil)
+		if err != nil {
+			t.Errorf("cannot GET Parameter by Name: %v - %v", parameters.Name, err)
+		}
+		if reqInf.StatusCode != http.StatusOK {
+			t.Errorf("Expected 200 status code, got %v", reqInf.StatusCode)
+		}
+		if len(resp) <= 0 {
+			t.Errorf("No data available for Get Parameters by Name")
+		}
+	}
+}
+
+func GetParametersByInvalidId(t *testing.T) {
+	resp, _, err := TOSession.GetParameterByID(10000, nil)
+	if err != nil {
+		t.Errorf("Getting Parameters by Invalid ID %v", err)
+	}
+	if len(resp) >= 1 {
+		t.Errorf("Invalid ID shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func GetParametersByInvalidName(t *testing.T) {
+	resp, _, err := TOSession.GetParametersByName("abcd", nil)
+	if err != nil {
+		t.Errorf("Getting Parameters by Invalid Name %v", err)
+	}
+	if len(resp) >= 1 {
+		t.Errorf("Invalid name shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func GetParametersByInvalidConfigfile(t *testing.T) {
+	resp, _, err := TOSession.GetParametersByConfigFile("abcd", nil)
+	if err != nil {
+		t.Errorf("Getting Parameters by Invalid ConfigFile %v", err)
+	}
+	if len(resp) >= 1 {
+		t.Errorf("Invalid config file shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func GetParametersByInvalidValue(t *testing.T) {
+	resp, _, err := TOSession.GetParametersByValue("abcd", nil)
+	if err != nil {
+		t.Errorf("Getting Parameters by Invalid value %v", err)
+	}
+	if len(resp) >= 1 {
+		t.Errorf("Invalid value shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func CreateTestParametersAlreadyExist(t *testing.T) {
+	resp, _, _ := TOSession.GetParameters(nil, nil)
+	if len(resp) > 0 {
```

This test can silently pass if Traffic Ops fails to behave as expected

---

Added failure checks for a response.